zanata-war/src/main/java/org/zanata/rest/RestLimitingSynchronousDispatcher.java

```diff
@@ -45,18 +71,33 @@ public RestLimitingSynchronousDispatcher(
         this.processor = processor;
     }
 
+    HttpServletRequest getServletRequest() {
+        return ServletContexts.instance().getRequest();
+    }
+
     @Override
     public void invoke(final HttpRequest request, final HttpResponse response) {
 
+        //This is for REST request from browser where user is logged in.
         HAccount authenticatedUser = getAuthenticatedUser();
```

Comment:
So at this point `authenticatedUser` is only non-null if the request came from a browser?  Could we say that?